### ISSUE TITLE
Conclusions?

### ISSUE THREAD

**Issue by jacereda** [floooh/sokol-zig#1]
Can you share your conclusions so far? Is there anything you miss in zig? Why did you need those `conv_xxx()` adapter functions? 

I'm considering starting a project using vulkan and I guess I'll face similar problems...

**Comment by floooh** [comment]
Hi, I'm currently waiting on this bug to be resolved (only seems to happen on macOS though): https://github.com/ziglang/zig/issues/3211

The reason why I duplicated some structs in the Zig manually, along with those *conv_* functions is mainly that I want the Zig wrapper module feels like it's been written in Zig from the outside. That means it should not expose the module user to any C types (like C strings or C functions). So structs which use such C types need a Zig 'mirror struct', and a conversion function which converts C types to Zig types.

But thankfully this isn't needed in most cases since most C types map identically to Zig types.

So in conclusion: the reason why there's a "manually augmented" Zig module instead of directly exposing the automatically imported C interface to the user is that I want to API user to feel like using a native Zig module, not an imported C module :)

**Comment by jacereda** [comment]
Got it, thanks.
